specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2021-05-31-privatepreview/Migrations.json

```diff
@@ -0,0 +1,818 @@
+    },
+    "MigrationSubState": {
+      "enum": [
+        "PerformingPreRequisiteSteps",
+        "WaitingForLogicalReplicationSetupRequestOnSourceDB",
+        "WaitingForDBsToMigrateSpecification",
+        "WaitingForTargetDBOverwriteConfirmation",
+        "WaitingForDataMigrationScheduling",
+        "WaitingForDataMigrationWindow",
+        "MigratingData",
+        "WaitingForCutoverTrigger",
+        "CompletingMigration",
+        "Completed"
+      ],
+      "type": "string",
+      "description": "Migration sub state.",
+      "x-ms-enum": {
+        "name": "MigrationSubState",
+        "modelAsString": true
+      }
+    },
+    "MigrationSubStateDetails": {
+      "type": "object",
+      "properties": {
+        "currentSubState": {
+          "$ref": "#/definitions/MigrationSubState",
+          "readOnly": true
+        }
+      },
+      "description": "Migration sub state details."
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the sku, typically, tier + family + cores, e.g. Standard_D4s_v3."
+        },
+        "tier": {
+          "type": "string",
+          "description": "The tier of the particular SKU, e.g. Burstable.",
+          "enum": [
+            "Burstable",
+            "GeneralPurpose",
+            "MemoryOptimized"
+          ],
+          "x-ms-enum": {
+            "name": "SkuTier",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "name",
+        "tier"
+      ],
+      "description": "Sku information related properties of a server."
+    },
+    "MigrationNameAvailabilityReason": {
+      "enum": [
+        "Invalid",
+        "AlreadyExists"
+      ],
+      "type": "string",
+      "description": "Migration name availability reason.",
+      "x-ms-enum": {
+        "name": "MigrationNameAvailabilityReason",
+        "modelAsString": true
+      }
+    },
+    "MigrationNameAvailabilityResource": {
+      "required": [
+        "name",
+        "type"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The resource name to verify."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the resource."
+        },
+        "nameAvailable": {
+          "type": "boolean",
+          "description": "Indicates whether the resource name is available.",
+          "readOnly": true
+        },
+        "reason": {
+          "$ref": "#/definitions/MigrationNameAvailabilityReason",
+          "description": "Migration name availability reason",
+          "readOnly": true
+        },
+        "message": {
+          "type": "string",
+          "description": "Migration name availability message.",
+          "readOnly": true
+        }
+      },
+      "description": "Represents a migration name's availability."
+    },
+    "CloudError": {
```

Comment:
Could you use  the ErrorResponse in v2/types.json instead ?

Reply:
We use this object across our api's see Microsoft.DBforPostgreSQL/stable/2021-06-01/postgresql.json